### PR TITLE
Bz2

### PR DESCRIPTION
**Pull request by Lcch**

### CODE REVIEW

File: examples/ch_wash/main.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"regexp"
+	"strings"
+	"unicode"
+
+	"github.com/naturali/kmr/executor"
+	kmrpb "github.com/naturali/kmr/pb"
+)
+
+var (
+	WASHER_SPLIT_PATTERN = `,|\.|!|\?|，|。|！|？|:|：|;|；|「|」|．|\t|：…｛｝`
```
Comment (xiang90) [inline examples/ch_wash/main.go]: do not user underscore style in go.

File: examples/ch_wash/main.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	WASHER_IGNORE_PUNCTS = " 　'\"《》‘’“”・-_<>〃〈〉()（）……@、【】[]*-、『』~"
+
+	ignorePuncts map[rune]bool
+)
+
+func is_alpha_or_number(r rune) bool {
```
Comment (xiang90) [inline examples/ch_wash/main.go]: isAlphaOrNumber

File: examples/ch_wash/main.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	return 'a' <= r && r <= 'z' || 'A' <= r && r <= 'Z' || unicode.IsDigit(r)
+}
+
+func is_chinese(r rune) bool {
+	return r >= '\u4e00' && r <= '\u9fa5'
+}
+
+func is_ignore_puncts(r rune) bool {
+	if _, ok := ignorePuncts[r]; ok {
```
Comment (xiang90) [inline examples/ch_wash/main.go]: _, ok:= ignorePuncts[r]
return ok

File: examples/ch_wash/main.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	re, _ := regexp.Compile(`^https?://.*[\r\n]*`)
+	line = re.ReplaceAllString(line, "")
+	return line
+}
+
+func process_single_sentence(line string) []string {
+	out := make([]string, 0)
+	e_word := ""
+	for _, r := range line {
+		if is_ignore_puncts(r) {
```
Comment (xiang90) [inline examples/ch_wash/main.go]: use switch instead.

File: examples/ch_wash/main.go (hunk 1 above)
Comment (Lcch) [inline examples/ch_wash/main.go]: done.

File: examples/ch_wash/main.go (hunk 2 above)
Comment (Lcch) [inline examples/ch_wash/main.go]: done

File: examples/ch_wash/main.go (hunk 3 above)
Comment (Lcch) [inline examples/ch_wash/main.go]: done

File: examples/ch_wash/main.go (hunk 4 above)
Comment (Lcch) [inline examples/ch_wash/main.go]: There is a bunch of function calls to determine what "r" is. How to use switch?

File: examples/ch_wash/main.go [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+func isSplitPuncts(r rune) bool {
+	if _, ok := splitPuncts[r]; ok {
+		return true
+	}
+	return false
+}
+
+func remove_illegal_pattern(line string) string {
```
Comment (xiang90) [inline examples/ch_wash/main.go]: removeIllegalPattern

File: examples/ch_wash/main.go [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+	if strings.HasPrefix(line, "<docno>") || strings.HasSuffix(line, "<url>") {
+		return ""
+	}
+	line = strings.Replace(strings.Replace(line, "</a>", "", -1), "<a>", "", -1)
+	re, _ := regexp.Compile(`^https?://.*[\r\n]*`)
+	line = re.ReplaceAllString(line, "")
+	return line
+}
+
+func process_single_sentence(line string) []string {
```
Comment (xiang90) [inline examples/ch_wash/main.go]: processSingleSentence

**Comment by xiang90** [comment]
@Lcch 

Please clean up all underscore style naming in the pr before merging.

File: examples/ch_wash/main.go (hunk 4 above)
Comment (xiang90) [inline examples/ch_wash/main.go]:
```
switch {
    case func(x):
    case func(y):
}
```

**Pull request merged**